src/esp/sim/BatchReplayRenderer.cpp

```diff
@@ -54,7 +54,17 @@ BatchReplayRenderer::BatchReplayRenderer(
   }
 }
 
-BatchReplayRenderer::~BatchReplayRenderer() = default;
+BatchReplayRenderer::~BatchReplayRenderer() {
+  doClose();
+}
+
+void BatchReplayRenderer::doClose() {
+  for (int i = 0; i < envs_.size(); ++i) {
+    envs_[i].player_.close();
+  }
+  envs_ = {};
+  renderer_.reset();
```

Comment:
Can you point me to this reset implementation? At a glance, I couldn't find it in the codebase.

Reply:
That's a shared pointer reset which aims to hit `RendererStandalone::~RendererStandalone()` if the renderer is standalone to release the graphics context.

Reply:
Ah, I see, reset on the smart pointer.